scripts/lib/CIME/XML/env_test.py

```diff
@@ -46,62 +46,57 @@ def set_test_parameter(self, name, value):
         otherwise create a node and initialize it to value
         """
         case = self.get_value("TESTCASE")
-        tnode = self.get_node("test",{"NAME":case})
-        # This xpath statement gets immediate children only (ignoring
-        # children of build and run blocks)
-        idnode = self.get_optional_node(name, root=tnode, xpath=name)
```

Comment:
No need for xpath, looking at immediate children is exactly what get_child does now.